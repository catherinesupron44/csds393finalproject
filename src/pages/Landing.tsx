import { Trophy, Users, Medal } from 'lucide-react';

interface LandingProps {
  onGetStarted: () => void;
}

export default function Landing({ onGetStarted }: LandingProps) {
  return (
    <div className="container mx-auto px-4 py-16 text-center">
      <h1 className="text-5xl font-bold text-indigo-900 mb-6">
        Welcome to BetBuddy
      </h1>
      <p className="text-xl text-gray-600 mb-8 max-w-2xl mx-auto">
        Track friendly bets, compete with friends, and climb the leaderboard - all without real money stakes!
      </p>
      <button
        onClick={onGetStarted}
        className="bg-indigo-600 text-white px-8 py-3 rounded-lg text-lg font-semibold hover:bg-indigo-700 transition-colors"
      >
        Get Started
      </button>
      
      <div className="grid grid-cols-1 md:grid-cols-3 gap-8 mt-16">
        <FeatureCard 
          icon={<Trophy className="w-8 h-8 text-indigo-600" />}
          title="Compete & Win"
          description="Create custom bets and challenge your friends in various categories"
        />
        <FeatureCard 
          icon={<Users className="w-8 h-8 text-indigo-600" />}
          title="Join Groups"
          description="Create or join betting groups based on your interests"
        />
        <FeatureCard 
          icon={<Medal className="w-8 h-8 text-indigo-600" />}
          title="Earn Badges"
          description="Unlock achievements and climb the global leaderboard"
        />
      </div>
    </div>
  );
}

function FeatureCard({ icon, title, description }: { 
  icon: React.ReactNode;
  title: string;
  description: string;
}) {
  return (
    <div className="bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-shadow">
      <div className="flex justify-center mb-4">
        {icon}
      </div>
      <h3 className="text-xl font-semibold text-gray-900 mb-2">{title}</h3>
      <p className="text-gray-600">{description}</p>
    </div>
  );
}