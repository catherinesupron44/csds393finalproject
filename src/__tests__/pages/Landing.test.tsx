import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import Landing from '../../pages/Landing';


/**
 * @jest-environment node
 */


describe('Landing Page', () => {
  const mockOnGetStarted = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders landing page content', () => {
    render(<Landing onGetStarted={mockOnGetStarted} />);
    
    expect(screen.getByText('Welcome to BetBuddy')).toBeInTheDocument();
    expect(screen.getByText('Get Started')).toBeInTheDocument();
    expect(screen.getByText('Compete & Win')).toBeInTheDocument();
    expect(screen.getByText('Join Groups')).toBeInTheDocument();
    expect(screen.getByText('Earn Badges')).toBeInTheDocument();
  });

  it('calls onGetStarted when button is clicked', async () => {
    render(<Landing onGetStarted={mockOnGetStarted} />);
    
    await userEvent.click(screen.getByText('Get Started'));
    expect(mockOnGetStarted).toHaveBeenCalledTimes(1);
  });
});