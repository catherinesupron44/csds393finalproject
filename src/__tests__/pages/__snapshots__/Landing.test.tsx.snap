// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto px-4 py-16 text-center"
  >
    <h1
      class="text-5xl font-bold text-indigo-900 mb-6"
    >
      Welcome to BetBuddy
    </h1>
    <p
      class="text-xl text-gray-600 mb-8 max-w-2xl mx-auto"
    >
      Track friendly bets and compete with friends - all without real money stakes!
    </p>
    <button
      class="bg-indigo-600 text-white px-8 py-3 rounded-lg text-lg font-semibold hover:bg-indigo-700 transition-colors"
    >
      Get Started
    </button>
    <div
      class="grid grid-cols-1 md:grid-cols-3 gap-8 mt-16"
    >
      <div
        class="bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-shadow"
      >
        <div
          class="flex justify-center mb-4"
        >
          <div
            data-testid="trophy-icon"
          >
            Trophy
          </div>
        </div>
        <h3
          class="text-xl font-semibold text-gray-900 mb-2"
        >
          Track Your Wins
        </h3>
        <p
          class="text-gray-600"
        >
          Effortlessly track your winnings and losses as you rise to the top.
        </p>
      </div>
      <div
        class="bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-shadow"
      >
        <div
          class="flex justify-center mb-4"
        >
          <div
            data-testid="users-icon"
          >
            Users
          </div>
        </div>
        <h3
          class="text-xl font-semibold text-gray-900 mb-2"
        >
          Join Exciting Markets
        </h3>
        <p
          class="text-gray-600"
        >
          Get in on the action by creating or joining markets that match your interests.
        </p>
      </div>
      <div
        class="bg-white p-6 rounded-xl shadow-lg hover:shadow-xl transition-shadow"
      >
        <div
          class="flex justify-center mb-4"
        >
          <div
            data-testid="star-icon"
          >
            Star
          </div>
        </div>
        <h3
          class="text-xl font-semibold text-gray-900 mb-2"
        >
          Bet, Compete & Win
        </h3>
        <p
          class="text-gray-600"
        >
          Challenge friends, place your bets, and prove youâ€™ve got the best instincts.
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
